### PR TITLE
[rest][new] Added s3 delete object action, fixed bucket removal after…

### PR DESCRIPTION
**Pull request by jstastny-cz**
//test: @s3,@file-transfer

FUSEQE-1943
FUSEQE-1944

### CODE REVIEW

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by jstastny-cz** [comment]
I did provide tags with escaped \@ sign :-( that caused the failure of the test job.

**Comment by jstastny-cz** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by tplevko** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by jstastny-cz** [comment]
Seems like there's an issue with passing two tags through the PR description:
> java.lang.IllegalArgumentException: Not a file or directory: /home/hudson/workspace/syndesis-tests-PR/syndesis-qe/rest-tests/file-transfer,@smoke

@tplevko how should I run the tests then?

**Comment by mcada** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: rest-tests/src/test/java/io/syndesis/qe/rest/tests/integrations/S3Steps.java [hunk 1]
```diff
@@ -1,5 +1,9 @@
 package io.syndesis.qe.rest.tests.integrations;
 
+import cucumber.api.java.After;
+import cucumber.api.java.Before;
+import io.syndesis.qe.endpoints.TestSupport;
+import io.syndesis.qe.utils.S3Utils;
```
Comment (mcada) [inline rest-tests/src/test/java/io/syndesis/qe/rest/tests/integrations/S3Steps.java]: remove unused imports

File: rest-tests/src/test/java/io/syndesis/qe/rest/tests/integrations/S3Steps.java [hunk 2]
```diff
@@ -42,12 +46,20 @@ public S3Steps() {
...
+
+    @Given("^create S3 polling START action step with bucket: \"([^\"]*)\" and prefix \"([^\"]*)\"$")
+    public void createS3PollingStep(String bucketName, String prefix) {
         final Connector s3Connector = connectorsEndpoint.get("aws-s3");
         final Connection s3Connection = connectionsEndpoint.get(S3BucketNameBuilder.getBucketName(bucketName));
         final Action s3PollingAction = TestUtils.findConnectorAction(s3Connector, "aws-s3-polling-bucket-connector");
         final Map<String, String> properties = TestUtils.map(TestUtils.map("deleteAfterRead", "false",
                 "maxMessagesPerPoll", "10",
                 "delay", "1000"));
+        if (prefix!=null){
```
Comment (mcada) [inline rest-tests/src/test/java/io/syndesis/qe/rest/tests/integrations/S3Steps.java]: run some kind of auto-layout on your changes so we have a nice code: (prefix != null)

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/S3ValidationSteps.java [hunk 3]
```diff
@@ -5,13 +5,15 @@
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 
+import cucumber.api.java.After;
 import cucumber.api.java.en.Given;
 import cucumber.api.java.en.Then;
 import io.syndesis.qe.endpoints.TestSupport;
```
Comment (mcada) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/S3ValidationSteps.java]: remove unused imports

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/S3ValidationSteps.java [hunk 4]
```diff
@@ -31,22 +33,9 @@ public S3ValidationSteps() {
...
-        TestSupport.getInstance().resetDB();
-    }
-
-    @Then("^clean S3, remove sample bucket with name: \"([^\"]*)\"")
-    public void cleanupS3bucket(String bucket) {
-        String bucketName = S3BucketNameBuilder.getBucketName(bucket);
-        log.debug("deleting bucket with name: {}", bucketName);
-        s3Utils.deleteS3Bucket(bucketName);
-        TestSupport.getInstance().resetDB();
+    @After("@cleanup-s3-after")
```
Comment (mcada) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/S3ValidationSteps.java]: this belongs into after hooks, for rest it is file: `RestTestHooks`. Lets keep it clear, it is not a validation step

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/S3ValidationSteps.java [hunk 5]
```diff
@@ -65,4 +54,38 @@ public void validateIntegration(String bucketName, String fileName, String text)
...
+        checkBucketsPresence(bucketNames, false);
+    }
+
+    /**
+     * Method that checks presence of given buckets in S3 instance.
+     * @param bucketNames name of the buckets to check
+     * @param shouldExist expected state to be checked against, true - should exist, false - shouldn't exist
+     */
+    private void checkBucketsPresence(List<String> bucketNames, boolean shouldExist){
+        SoftAssertions softly = new SoftAssertions();
```
Comment (mcada) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/S3ValidationSteps.java]: Nice!

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Pull request merged**